src/Alley/WP/find-result.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * `find_result()` function
+ *
+ * (c) Alley <info@alley.co>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @package wp-find-one
+ */
+
+namespace Alley\WP;
+
+/**
+ * Returns the first item in the given array if the item is an instance of a
+ * specific class or is a non-object.
+ *
+ * @param string $class   If the result plucked from $results is an object, the
+ *                        name of the class the object must be an instance of
+ *                        for it to be returned.
+ * @param array  $results The array of results from the WordPress query
+ *                        function to retrieve the result from.
+ * @return mixed|null An object of the given $class; a non-object if the item
+ *                        plucked from $results is not an object; null if no
+ *                        result is found.
+ */
+function find_result( string $class, $results ) {
+	if ( ! $results || ! \is_array( $results ) ) {
+		return null;
+	}
+
```

Comment:
I'm question whether this low-level function's responsibility to perform silently-failing safety checks for `$results`. The fatal error resulting from an incorrect parameter type would be much more informative in a "fail fast and fail bad" kind of way.

Without the certainty provided by a fatal error, an unexpectedly-not-an-array input could pass through this conditional gate and all you would get as a result would be an unexpectedly-null output, spending the next few minutes opening up the debugger to figure out what happened.

```suggestion
function find_result( string $class, array $results ) {
```

Reply:
I decided to sleep on this suggestion, but I'm still inclined to omit the parameter type declaration.

First, these are meant to be convenience functions, and requiring that the caller verify the type before using this function makes it less convenient. Consider an API or function that returns `null` or `false` rather than an empty set under normal operations — it wouldn't be possible to pass the results of those calls directly to `find_result()`.

Second, as mentioned earlier, the `find_one_*()`  wrapper functions represent the overwhelming (exclusive?) usage of `find_result()`, so even if `find_result()` was strictly typed, the wrapper functions would swallow the non-array results before the caller was aware of them.

Third, if the caller has to check `is_array( $results )`, they might as well check `! empty( $results[0] )` instead, at which point `find_result()` isn't providing much value.

If it's important for the caller to know whether the results are an array and why, it's still possible to do so before using this function, but, also, maybe these functions aren't going to be right for those use cases.